variables.tf

```diff
@@ -271,6 +284,37 @@ EOF
 
 }
 
+variable "internal_lb_ip" {
```

Comment:
Why is this a variable? Shouldn't there be a `google_compute_address` resource, roughly corresponding to the external IP address resource?

So the address part of `network.tf` should look something like:

```
resource "google_compute_address" "vault" {
  count   = local.use_external_lb ? 1 : 0
  project = var.project_id	  project = var.project_id

  name   = "vault-lb"	  name   = "vault-lb"
  region = var.region	  region = var.region
  depends_on = [google_project_service.service]	  depends_on = [google_project_service.service]
}

resource "google_compute_address" "vault" {
  count   = local.use_external_lb ? 0 : 1
  project = var.project_id

  name         = "vault-lb-internal"
  region       = var.region
  address_type = "INTERNAL"
  subnetwork   = local.subnet

  depends_on = [google_project_service.service]
}
```

Then make the corresponding change from `var` to `resource` in `lb_ip`.

Reply:
Hi there, thanks for the comments. Take a look at this comment thread for some more context. https://github.com/terraform-google-modules/terraform-google-vault/pull/84#pullrequestreview-350504154 

A `google_compute_address` is a public IP, which for obvious reasons cannot be assigned to an internal load balancer. Due to the fact that we need access to the LB IP in the startup script for Vault config, we have to hard code this value as opposed to letting the load balancer create it.

Reply:
`google_compute_address` has an `address_type` field, which can be set to `INTERNAL` to pick a private, RFC 1918 IP address.

Reply:
Good to know. This seems like a reasonable addition.